src/components/node-list/index.js

```diff
@@ -87,41 +102,64 @@ const NodeListProvider = ({
   const onItemMouseLeave = (item) => {
     if (isTagType(item.type)) {
       onToggleTagActive(item.id, false);
+    } else if (isModularPipelineType(item.type)) {
+      onToggleModularPipelineActive(item.id, false);
     } else if (item.visible) {
       onToggleNodeActive(null);
     }
   };
 
   const onToggleGroupChecked = (type, checked) => {
-    if (isTagType(type)) {
-      // Filter all tags if at least one tag item set, otherwise enable all tags
-      const tagItems = items[type] || [];
-      const someTagSet = tagItems.some((tagItem) => !tagItem.unset);
-      const allTagsValue = someTagSet ? undefined : true;
-      onToggleTagFilter(
-        tagItems.map((tag) => tag.id),
-        allTagsValue
+    if (isTagType(type) || isModularPipelineType(type)) {
+      // Filter all category items if at least one item set, otherwise enable all items
+      const categoryItems = items[type] || [];
+      const someCategoryItemSet = categoryItems.some(
+        (categoryItem) => !categoryItem.unset
       );
+      const allCategoryItemsValue = someCategoryItemSet ? undefined : true;
+
+      if (isTagType(type)) {
+        onToggleTagFilter(
+          categoryItems.map((tag) => tag.id),
+          allCategoryItemsValue
+        );
+      } else {
+        onToggleModularPipelineFilter(
+          categoryItems.map((item) => item.id),
+          allCategoryItemsValue
+        );
+      }
     } else {
       onToggleTypeDisabled(type, checked);
     }
   };
 
-  const onTagItemChange = (tagItem, wasChecked) => {
-    const tagItems = items[tagItem.type] || [];
-    const oneTagChecked =
-      tagItems.filter((tagItem) => tagItem.checked).length === 1;
-    const shouldResetTags = wasChecked && oneTagChecked;
-
-    if (shouldResetTags) {
-      // Unset all tags
-      onToggleTagFilter(
-        tags.map((tag) => tag.id),
-        undefined
-      );
+  const onCategoryItemChange = (item, wasChecked) => {
+    const categoryType = item.type;
+    const categoryTypeItems = items[categoryType] || [];
+    const oneCategoryItemChecked =
+      categoryTypeItems.filter((categoryItem) => categoryItem.checked)
+        .length === 1;
+    const shouldResetCategoryItems = wasChecked && oneCategoryItemChecked;
+
+    if (shouldResetCategoryItems) {
+      // Unset all category item
+      if (categoryType === 'tag') {
+        onToggleTagFilter(
+          tags.map((tag) => tag.id),
+          undefined
+        );
+      } else {
+        onToggleModularPipelineFilter(
+          modularPipelines.map((modularPipeline) => modularPipeline.id),
+          undefined
+        );
+      }
     } else {
-      // Toggle the tag
-      onToggleTagFilter([tagItem.id], !wasChecked);
+      // Toggle the category item
+      categoryType === 'tag'
```

Comment:
I think since this is just calling functions rather than using their return values, better to make this an if-else block (like used a few lines above) rather than use the ternary operator here.